src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGRStrategies.hpp

```diff
@@ -52,7 +52,7 @@ namespace mgr
  */
 template< typename STRATEGY >
 void setStrategy( LinearSolverParameters::MGR const & params,
-                  arrayView1d< localIndex const > const & numComponentsPerField,
+                  arrayView1d< integer const > const & numComponentsPerField,
```

Comment:
Hi @klevzoff !

I notice that you change `localIndex` to `integer` in your patch.
I always thought that behind the naming `localIndex` and `globalIndex` there was some kind of idea that those indices we intrinsically not the same kind of indices: some local to the MPI rank, some global to the simulation.

In some sense, it was like we were using some Hungarian notation for indices. (Not exactly since we use it on types and not on variable names, but there is some flavor).

(It's true that here `numComponentsPerField` is not indeed an _index_.)

We had spoken about having some `unit` system inside of GEOSX.
Do you think this could also be a use case for this `unit` stuff?

poke @rrsettgast

Reply:
Hi @TotoGaz ! What you describe for `localIndex` vs `globalIndex` is true. Using either of these types to represent number of components per degree-of-freedom (or component _index_ for that matter) doesn't feel right, because it doesn't scale with number of mesh objects (either local or global), even though it often gets added to those things to form a new (local or global) index. It was a historical mistake of me misusing `localIndex` for all kinds of things where I shouldn't have. 

Doing what you suggest in my mind implies introducing strong types for indices, and while it's not a bad idea per se, I imagine this being a huge headache in terms of implementation, especially considering that index arrays are part of interop with TPLs (e.g. directly handed off to linear algebra packages).

Reply:
> Doing what you suggest in my mind implies introducing strong types for indices, and while it's not a bad idea per se, I imagine this being a huge headache in terms of implementation, especially considering that index arrays are part of interop with TPLs (e.g. directly handed off to linear algebra packages).

Indeed that may be an issue.
There was this idea to `typedef` our type to `unit<int>` or `int` based on some compilation flag.
Maybe there are some `unit` libs that allow some automatic conversion from their `unit<int>` to `int` (maybe an option? Maybe `units::no_dim<int> -> int` ?). But what about pointers to data ? This needs to be investigated as part of the more global discussion on `units`.